src/services/formatting/smartIndenter.ts

```diff
@@ -729,6 +730,12 @@ export namespace SmartIndenter {
                     return false;
                 }
                 break;
+            case SyntaxKind.TryStatement:
+                const tryStatement = parent as TryStatement;
+                if (tryStatement.finallyBlock && tryStatement.finallyBlock === child) {

```

Comment:
I'm suspicious about this just because all of the other code in this switch case are only checking syntax kind or position info, not node equality.

Reply:
@jakebailey thanks for the feedback. I found that in this case, the `finallyBlock` should not be handled as a child of the try block to ensure the correct delta calculation 

https://github.com/microsoft/TypeScript/blob/5170645f4e4b395891fad823bfe03ac628411afb/src/services/formatting/formatting.ts#L624

There are other places where we handle the `finallyBlock` in a similar way

https://github.com/microsoft/TypeScript/blob/5170645f4e4b395891fad823bfe03ac628411afb/src/services/outliningElementsCollector.ts#L259-L261

I hope that makes sense., if you have any thoughts or suggestions, I’d be grateful to hear them. thanks

Reply:
How does this bug not affect the try block or the catch block, then? Maybe I'm missing something...

Reply:
I've modified changes to rely exclusively on checking the syntax kind